ntbs-service/Pages/Search/Index.cshtml.cs

```diff
@@ -75,14 +73,11 @@ public async Task<IActionResult> OnGetAsync(int? pageIndex)
                 .FilterBySex(SearchParameters.SexId)
                 .GetResult();
 
-            IQueryable<Notification> notificationIdsQueryable = searchService.OrderQueryableByNotificationDate(filteredDrafts)
-                                                                .Union(searchService.OrderQueryableByNotificationDate(filteredNonDrafts));
-
-            var notificationIds = await searchService.GetPaginatedItemsAsync(notificationIdsQueryable.Select(n => n.NotificationId), PaginationParameters);
-            var count = await notificationIdsQueryable.CountAsync();
-            IEnumerable<Notification> notifications = await notificationService.GetNotificationsByIdAsync(notificationIds);
+            KeyValuePair<IList<int>, int> notificationIdsAndCount = await searchService.UnionAndPaginateQueryables(filteredDrafts, filteredNonDrafts, PaginationParameters);

```

Comment:
Using a [`Tuple`](https://docs.microsoft.com/en-us/dotnet/api/system.tuple-2?view=netcore-2.2) couldmake sense here, but a KeyValuePair is drawing the wrong picture!
So either tuple, or an explicit object/out params. (And i think i'm leaning towards the latter for readability, as long as they're easy to work with in the test, too)
Sorry if my use of the word "Pair" lead you astray here!